core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -90,14 +103,18 @@ private enum Codec {
   }
 
   public static WriteBuilder write(OutputFile file) {
+    if (file instanceof EncryptedOutputFile) {
```

Comment:
Encryption is handled by adding this. The `read` side already has a similar check.